admin/services/data-access/group.data.service.js

```diff
@@ -20,17 +24,99 @@ groupDataService.getGroups = async function (query) {
     .exec()
 }
 
+/**
+ * Get active groups (non-soft-deleted).
+ * @returns {Promise<*>}
+ */
+groupDataService.sqlFindGroups = async (schoolId) => {
+  const sql = `
+  SELECT g.id, g.name, COUNT(pg.pupil_id) as pupilCount 
+  FROM ${sqlService.adminSchema}.[group] g
+  LEFT OUTER JOIN ${sqlService.adminSchema}.pupilGroup pg 
+  ON g.id = pg.group_id
+  WHERE g.isDeleted=0
+  AND g.school_id=@schoolId
```

Comment:
school_id doesn't seem to appear in the diff for the CREATE TABLE [group].

Reply:
new script as discussed